### ISSUE TITLE
opt out old-school mixin-like classes from prefer_mixin

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#2514]
Flutter has some classes (like ChangeNotifier and WidgetsBindingObserver) where we tell developers that those can be extended, implemented, or mixed-in. However, the `prefer_mixin` lint is not happy when they are used as a mixin in a `with` clause. Since those classes pre-date the introduction of actual mixins in Dart, it would be nice if they can be opted out from the lint, e.g. by annotating them with something like `@legacyMixin`.

The Dart SDK also has some of these legacy mixins that would benefit from such an annotation (example: IterableMixin) to avoid the `prefer_mixin` warning in the following code snippet:

```dart
/// info: Prefer using mixins. (prefer_mixin)
abstract class Foo with IterableMixin<int> {
  // ...
}
```

/cc @pq @Hixie @mit-mit @munificent @devoncarew

**Comment by bwilkerson** [comment]
If the annotation were to be used in the SDK it would need to be defined in the SDK, so I'll cc a couple more people: @leafpetersen @lrhn.
